### PR TITLE
Send events to Kafka instead of Kinesis

### PR DESCRIPTION
**Pull request by spladug**
This points the injectors at Kafka instead of Kinesis and tears out the batching stuff since we want individual records in Kafka and kafka-python can handle request batching for us.

I've tested this out locally and it seems to work well, but we'll need to test it at scale to get a real feel for it.

:eyeglasses: @dellis23 @bsimpson63 

cc: @Kaitaan

### CODE REVIEW

File: events/injector.py [hunk 1]
```diff
@@ -169,15 +39,39 @@ def main():
...
-    region = config["aws.region"]
-    kinesis = boto.kinesis.connect_to_region(region)
 
-    metrics_client = baseplate.make_metrics_client(config)
-    kinesis_batch_consumer = KinesisBatchConsumer(
-        kinesis, topic_name, metrics_client)
-    batcher = Batcher(kinesis_batch_consumer)
-    consume_items_in_batches(messages, batcher)
+    producer_options = {
+        "codec": CODEC_GZIP,
```
Comment (dellis23) [inline events/injector.py]: Will this save us even more in bandwidth, or is transfer free once you're inside the same AZ?

File: events/injector.py [hunk 2]
```diff
@@ -169,15 +39,39 @@ def main():
...
+            kafka_producer = SimpleProducer(kafka_client, **producer_options)
+        except KafkaError as exc:
+            _LOG.warning("could not connect: %s", exc)
+            metrics_client.counter("injector.connection_error").increment()
+            time.sleep(_RETRY_DELAY)
+            continue
+
+        for message in messages:
+            if message is not None:
+                for retry in itertools.count():
```
Comment (dellis23) [inline events/injector.py]: What's the behavior if Kafka fills up right now?  What happens if the local message queues fill up on the event collector?  Just a curiosity more than anything.

**Comment by dellis23** [comment]
:fish: I don't see any obvious problems.  I just have a few questions about the failure cases.  Cool to see less moving parts accomplishing the same thing!

File: events/injector.py (hunk 1 above)
Comment (spladug) [inline events/injector.py]: Free inside the same AZ.

File: events/injector.py (hunk 2 above)
Comment (spladug) [inline events/injector.py]: > What's the behavior if Kafka fills up right now?

If Kafka's unavailable, we'll retry until it comes back up. The IPC queue will fill in the meantime. In the future, we may decide to fallback to shipping to S3 or something during catostrophic outages.

> What happens if the local message queues fill up on the event collector?

If the local queue fills up, the collector starts returning 500s to clients. This'll mean lost messages from browsers etc. if they don't retry, but server-side events will just back up on r2 side.

**Comment by bsimpson63** [comment]
:fish:

**Comment by spladug** [comment]
:haircut: one additional commit which allows us to have multiple injector instances per queue for higher throughput.

**Comment by dellis23** [comment]
Hmm, I don't think my :fish: worked.

**Comment by bsimpson63** [comment]
:fish:

**Comment by spladug** [comment]
2cfe594fa3f6996687f919cf75ad19b36c573e25...469e50cbc0d217fda1087b9d62aefe4f10fa656b

**Pull request closed without merging**